index.js

```diff
@@ -86,45 +88,63 @@ exports.init = function (sbot, config) {
     sbot.publish(content, cb)
   }
 
+  function graph (cb) {
+    onReady(() => {
+      cb(null, layered.getGraph())
+    })
+  }
+
+  function graphStream (opts) {
+    opts = opts || {}
+    const live = opts.live === true // default is false
+    const old = opts.old !== false // default is true
+    if (live) {
+      return pCont((cb) => {
+        onReady(() => {
+          const unsubscribe = layered.onEdge((source, dest, value) => {
+            p.push({ source, dest, value })
+          })
+          const p = Pushable(unsubscribe)
+          if (old) {
+            p.push(layered.getGraph())
+          }
+          cb(null, p)
+        })
+      })
+    } else {
+      return pCont((cb) => {
+        onReady(() => {
+          cb(null, pull.once(layered.getGraph()))
+        })
+      })
+    }
+  }
+
   function hops (opts, cb) {
     if (typeof opts === 'function') {
       cb = opts
       opts = {}
     }
 
-    layered.onReady(() => {
-      if (sbot.db) {
-        sbot.db.onDrain('contacts', () => cb(null, layered.getHops(opts)))
-      } else {
-        cb(null, layered.getHops(opts))
-      }
+    onReady(() => {
+      cb(null, layered.getHops(opts))
     })
   }
 
-  // glue modules together
-  if (config.friends.hookAuth !== false) // defaults to true
-    authGlue(sbot, isBlocking)
-
-  if (config.friends.hookReplicate !== false) // defaults to true
-    replicationGlue(sbot, layered, legacy)
+  // Make sure blocked peers cannot connect, default is true
+  if (config.friends.hookAuth !== false) {
```

Comment:
There is no guarantee config.friends is defined!

:fire:

Reply:
Actually this is the only thing that I'm sure is defined. See the first line inside the implementation of `exports.init` function.

Reply:
Ah yes! The old monkey patch 🙊